app/services/participants/base.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+module Participants
+  class Base
+    include ProfileAttributes
+
+    class << self
+      def call(params:)
+        new(params: params).call
+      end
+
+      def not_implemented_error
+        raise NotImplementedError, "Method must be implemented"
+      end
+
+      delegate :valid_courses, :state_to_transition_to, to: :not_implemented_error
+    end
+
+    def call
+      unless valid?
+        raise ActionController::ParameterMissing, errors.map(&:message)
+      end
+
+      business_validation!
```

Comment:
Yes. I want to move that one, since the errors that it will give should be different when it's an invalid state transition as opposed to a guard against changing other things.